### PR TITLE
[FEATURE] Replace MetricFunctionTypes.IDENTITY domain type with convenience method get_domain_records() for SqlAlchemyExecutionEngine

### PR DESCRIPTION
**Pull request by alexsherstinsky**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
-
-
-

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
- JIRA: GREAT-56/GE-447
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/execution_engine/sqlalchemy_execution_engine.py [hunk 1]
```diff
@@ -372,28 +372,21 @@ def _get_sqlalchemy_key_pair_auth_url(
...
 
-    def get_compute_domain(
+    def get_domain_records(
         self,
         domain_kwargs: Dict,
-        domain_type: Union[str, MetricDomainTypes],
-        accessor_keys: Optional[Iterable[str]] = None,
-    ) -> Tuple[Select, dict, dict]:
-        """Uses a given batch dictionary and domain kwargs to obtain a SqlAlchemy column object.
+    ) -> Select:
```
Comment (jcampbell) [inline great_expectations/execution_engine/sqlalchemy_execution_engine.py]: Selectable?

File: great_expectations/execution_engine/sqlalchemy_execution_engine.py [hunk 2]
```diff
@@ -449,6 +440,128 @@ class MetricDomainTypes.
...
+            like to be using, or a corresponding string value representing it. String types include "identity",
+            "column", "column_pair", "table" and "other". Enum types include capitalized versions of these from the
+            class MetricDomainTypes.
+            accessor_keys (str iterable) - keys that are part of the compute domain but should be ignored when
+            describing the domain and simply transferred with their associated values into accessor_domain_kwargs.
+
+        Returns:
+            SqlAlchemy column
+        """
+        # Extracting value from enum if it is given for future computation
```
Comment (jcampbell) [inline great_expectations/execution_engine/sqlalchemy_execution_engine.py]: comment got detached

File: great_expectations/execution_engine/sqlalchemy_execution_engine.py [hunk 3]
```diff
@@ -449,6 +440,128 @@ class MetricDomainTypes.
...
+                    # noinspection PyPep8Naming
+                    column_B_name = domain_kwargs["column_B"]
+
+                ignore_row_if = domain_kwargs["ignore_row_if"]
+                if ignore_row_if == "both_values_are_missing":
+                    selectable = (
+                        sa.select([sa.text("*")])
+                        .select_from(selectable)
+                        .where(
+                            sa.and_(
```
Comment (jcampbell) [inline great_expectations/execution_engine/sqlalchemy_execution_engine.py]: you're reusing the column name in both places...which makes me think we must not have a test exercising this logic yet. Can you ensure we have a test in place to validate these cases?

File: great_expectations/execution_engine/sqlalchemy_execution_engine.py [hunk 4]
```diff
@@ -449,6 +440,128 @@ class MetricDomainTypes.
...
+                    )
+                elif ignore_row_if == "either_value_is_missing":
+                    selectable = (
+                        sa.select([sa.text("*")])
+                        .select_from(selectable)
+                        .where(
+                            sa.or_(
+                                sa.column(column_A_name) != None,
+                                sa.column(column_A_name) != None,
+                            )
```
Comment (jcampbell) [inline great_expectations/execution_engine/sqlalchemy_execution_engine.py]: I think this logic is wrong; it seems to be saying "keep if either value is present", which means if one value is missing and one present, the row is NOT ignored (backwards from the user requested behavior)

File: great_expectations/execution_engine/sqlalchemy_execution_engine.py [hunk 5]
```diff
@@ -449,6 +440,128 @@ class MetricDomainTypes.
...
+                    selectable = (
+                        sa.select([sa.text("*")])
+                        .select_from(selectable)
+                        .where(
+                            sa.or_(
+                                *(
+                                    sa.column(column_name) != None
+                                    for column_name in column_list
+                                )
+                            )
```
Comment (jcampbell) [inline great_expectations/execution_engine/sqlalchemy_execution_engine.py]: I think this is a logic error: it says *keep* if any value is *not* missing, but should say *ignore* if any value *is* missing.

File: great_expectations/execution_engine/sqlalchemy_execution_engine.py [hunk 6]
```diff
@@ -449,6 +440,128 @@ class MetricDomainTypes.
...
+                        sa.select([sa.text("*")])
+                        .select_from(selectable)
+                        .where(
+                            sa.and_(
+                                sa.column(column_A_name) != None,
+                                sa.column(column_A_name) != None,
+                            )
+                        )
+                    )
+                elif ignore_row_if == "either_value_is_missing":
```
Comment (jcampbell) [inline great_expectations/execution_engine/sqlalchemy_execution_engine.py:475]: Second test addition request: a test that shows proper sqlalchemy handling of *both* row_condition and ignore_row_if

**Review by jcampbell — CHANGES_REQUESTED** [review]

File: great_expectations/execution_engine/sqlalchemy_execution_engine.py (hunk 1 above)
Comment (alexsherstinsky) [reply great_expectations/execution_engine/sqlalchemy_execution_engine.py]: @jcampbell I changed this type to `Selectable` consistently throughout the `SqlAlchemyExecutionEngine` class/module.  Thanks.

File: great_expectations/execution_engine/sqlalchemy_execution_engine.py (hunk 2 above)
Comment (alexsherstinsky) [reply great_expectations/execution_engine/sqlalchemy_execution_engine.py]: @jcampbell Nice catch -- I placed the comment above the appropriate line.  Thanks.

File: great_expectations/execution_engine/sqlalchemy_execution_engine.py (hunk 3 above)
Comment (alexsherstinsky) [reply great_expectations/execution_engine/sqlalchemy_execution_engine.py]: @jcampbell Great catch!  This was an unfortunate typo.  I corrected it and added a test case to exercise the functionality.

File: great_expectations/execution_engine/sqlalchemy_execution_engine.py (hunk 4 above)
Comment (alexsherstinsky) [reply great_expectations/execution_engine/sqlalchemy_execution_engine.py]: @jcampbell You are right.  I fixed the typo in the second column name and also corrected the logic.  Thanks for catching this.

File: great_expectations/execution_engine/sqlalchemy_execution_engine.py (hunk 5 above)
Comment (alexsherstinsky) [reply great_expectations/execution_engine/sqlalchemy_execution_engine.py]: @jcampbell You are right. I corrected the logic and added a test case to exercise it. Thanks for catching this.

File: great_expectations/execution_engine/sqlalchemy_execution_engine.py (hunk 6 above)
Comment (alexsherstinsky) [reply great_expectations/execution_engine/sqlalchemy_execution_engine.py:475]: @jcampbell Please see the previous comments and the additional two test cases.  All of the "column_pair" and the "multicolumn" test cases utilize both, `row_condition` and `ignore_row_if` directives.

**Review by jcampbell — APPROVED** [review]
🎉

**Pull request merged**